drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -672,6 +700,36 @@ def _get_user_state(self):
             'overall_feedback': self._present_overall_feedback(overall_feedback_msgs)
         }
 
+    def _get_correct_state(self):
+        """
+        Returns one of the possible correct states for the configured data.
+        """
+        state = {}
+        items = copy.deepcopy(self.data.get('items', []))
+        for item in items:
+            zones = item.get('zones')
+
+            # For backwards compatibility
+            if zones is None:
+                zones = []
+                zone = item.get('zone')
+                if zone is not None and zone != 'none':
+                    zones.append(zone)
+
+            if len(zones):
```

Comment:
@arbrandes Hm. I'm probably missing something/not interpreting your comment correctly, but for the case where `zones` is `[]`, both `if zones:` and `if len(zones):` would evaluate to false, right? And once execution reaches this line (L719), `zones` can not be `None` (the `if` block that starts in L713 above takes care of that). So I don't see why it's necessary to do `if len(zones):` here yet. Please elaborate? :)